main.cpp

```diff
@@ -3930,7 +3929,15 @@ struct progress_bar {
         if (_percent != percent) {
             percent = _percent;
             unsigned int len = (width * percent) / 100;
-            std::cout << prefix << "[" << string(len, '=') << string(width-len, ' ') << "]  " << std::to_string(percent) << "%\r" << std::flush;
+            // Align all bars with the longest possible prefix string
+            auto longest_mem = std::max_element(
+                std::begin(memory_names), std::end(memory_names),
+                [] (const auto & p1, const auto & p2) {
+                    return p1.second.length() < p2.second.length();
+                }
+            );
```

Comment:
Could / should this calculation be done just once, rather than for _every_ time you call the `progress()` function?

Reply:
Yep, I can calculate it once when the progress_bar is created